JsonPatch/PatchExtensions.cs

```diff
@@ -7,136 +7,157 @@
+				var prop = mod.EnumerateObject().First(p => p.NameEquals(k));
+				patch.Add(PatchOperation.Add(path.Combine(k), prop.Value));
+			}
 
-            foreach (var k in origNames.Intersect(modNames))
-            {
-                var origProp = orig.EnumerateObject().First(p => p.NameEquals(k));
-                var modProp = mod.EnumerateObject().First(p => p.NameEquals(k));
+			foreach (var k in origNames.Intersect(modNames))
+			{
+				var origProp = orig.EnumerateObject().First(p => p.NameEquals(k));
+				var modProp = mod.EnumerateObject().First(p => p.NameEquals(k));
 
-                if (origProp.Value.ValueKind != modProp.Value.ValueKind)
-                {
-                    patch.Add(PatchOperation.Replace(JsonPointer.Parse(path + modProp.Name), modProp.Value));
-                }
-                else if (!string.Equals(origProp.Value.ToString(), modProp.Value.ToString()))
-                {
-                    if (origProp.Value.ValueKind == JsonValueKind.Object)
-                        PatchForObject(origProp.Value, modProp.Value, patch, path.Combine(modProp.Name));
-                    else if (origProp.Value.ValueKind == JsonValueKind.Array)
-                        PatchForArray(origProp.Value, modProp.Value, patch, path.Combine(modProp.Name));
-                    else
-                        patch.Add(PatchOperation.Replace(path.Combine(modProp.Name), modProp.Value));
-                }
-            }
-        }
-        private static void PatchForArray(JsonElement orig, JsonElement mod, List<PatchOperation> patch, JsonPointer path)
-        {
-            for (int i = 0; i < Math.Max(orig.GetArrayLength(), mod.GetArrayLength()); i++)
-            {
-                if (i >= orig.GetArrayLength())
-                {
-                    patch.Add(PatchOperation.Add(path.Combine(i), mod[i]));
-                    continue;
-                }
+				if (origProp.Value.ValueKind != modProp.Value.ValueKind)
+					patch.Add(PatchOperation.Replace(JsonPointer.Parse(path + modProp.Name), modProp.Value));
+				else if (!string.Equals(origProp.Value.ToString(), modProp.Value.ToString())) // TODO
+				{
+					if (origProp.Value.ValueKind == JsonValueKind.Object)
+						PatchForObject(origProp.Value, modProp.Value, patch, path.Combine(modProp.Name));
+					else if (origProp.Value.ValueKind == JsonValueKind.Array)
+						PatchForArray(origProp.Value, modProp.Value, patch, path.Combine(modProp.Name));
+					else
+						patch.Add(PatchOperation.Replace(path.Combine(modProp.Name), modProp.Value));
+				}
+			}
+		}
 
-                if (i >= mod.GetArrayLength())
-                {
-                    patch.Add(PatchOperation.Remove(path.Combine(i)));
-                    continue;
-                }
+		private static void PatchForArray(JsonElement orig, JsonElement mod, List<PatchOperation> patch, JsonPointer path)
+		{
+			for (int i = 0; i < Math.Max(orig.GetArrayLength(), mod.GetArrayLength()); i++)
+			{
+				var ui = (uint) i;
```

Comment:
Why not implicit int conversion? Here you have to make 2 casts. From int to uint, and then to string inside. Why not directly into to string? Just thoughts

Reply:
It's not proper for JSON Pointer to accept a negative number.

[RFC 6901 section 4](https://datatracker.ietf.org/doc/html/rfc6901#section-4)

> If the currently referenced value is a JSON array, the reference token MUST contain either:
> 
> - characters comprised of digits (see ABNF below; note that leading zeros are not allowed) that represent an **_unsigned base-10 integer value_**, making the new referenced value the array element with the zero-based index identified by the token, or
> - exactly the single character "-", making the new referenced value the (nonexistent) member after the last array element.

Reply:
Thanks for clarification. It's fair enough